Add heartbeating in exporter

splunkericl opened the pull request:
**Description:** <Describe what has changed.>
This change add configuration to enable heartbeating in splunkhecexporter. A heartbeat is an internal event log sent from exporter to destination. This even contains metadata about the current host like OS, build info, etc.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/19220

**Testing:**
- Updated unit tests for configuration, client and factory
- Added unit test for heartbeat module

**Documentation:** N/A

splunkericl commented:
@atoulme @dmitryax . After our discussions in the github issue, I am creating this draft PR to outline how I will implement it. No testing yet but if it looks good I will add tests and wrap it up.

atoulme commented on exporter/splunkhecexporter/factory.go:
```diff
@@ -171,6 +171,7 @@ func createLogsExporter(
 		telemetrySettings: set.TelemetrySettings,
 		buildInfo:         set.BuildInfo,
 	}
+	c.heartbeater = newHeartbeater(cfg, c.pushLogData)
```
you might want to also create one for the metric exporter?

splunkericl replied:
Right I should add them in traces and metric as well.

timannguyen commented:
i think you would need to update https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/splunkhecexporter/README.md with the associated heartbeat config.

splunkericl commented:
Discarded the comments and updated the repo to address conflicts. Moved to https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/20426
